src/Generator/DiffGenerator.php

```diff
@@ -112,42 +110,37 @@ private function createEmptySchema(): Schema
         return $this->emptySchemaProvider->createSchema();
     }
 
-    private function createFromSchema(): Schema
-    {
-        return $this->schemaManager->introspectSchema();
-    }
-
-    private function createToSchema(): Schema
+    /**
+     * Creates the schema from the database, ensuring tables from the target schema are whitelisted for comparison.
+     *
+     * @see https://github.com/doctrine/orm/pull/7875
+     */
+    private function createFromSchema(Schema $toSchema): Schema
     {
-        $toSchema = $this->schemaProvider->createSchema();
+        // backup schema assets filter
+        $previousFilter = $this->dbalConfiguration->getSchemaAssetsFilter();
 
-        $schemaAssetsFilter = $this->dbalConfiguration->getSchemaAssetsFilter();
+        if ($previousFilter === null) {
+            return $this->schemaManager->introspectSchema();
```

Comment:
This is kept for compatibility with DBAL v3. In DBAL v4, it can no longer be null.